pkg/cloud/cloud.go

```diff
@@ -187,9 +204,21 @@ func (c *cloud) CreateDisk(volumeName string, diskOptions *DiskOptions) (*Disk,
 		Tags:         tags,
 	}
 
-	m := c.GetMetadata()
+	var (
+		zone string
+		err  error
+	)
+	if diskOptions.AvailabilityZone == nil {
+		zone, err = c.pickRandomAvailabilityZone()
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		zone = *diskOptions.AvailabilityZone
+	}
```

Comment:
What about making `DiskOptions.AvailabilityZone` a `string` instead? Then you can pick a randon AZ if it it's an emtpy string.

Currently it's declared as a pointer, then we make it a string (through `zone`), then make it a pointer again (line 221 below).

Reply:
Good point. I gave it a second thought and I feels this is more clear that double nil and empty are the same invalid case and this remove the extra logic to do `stringPtr()`. updated the code.